src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/volume/v2/OpenStackVolumePluginTest.java

```diff
@@ -1,201 +1,524 @@
+            Assert.fail();
+        } catch (NoAvailableResourcesException e) {
+            // verify
+            Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).getPrefixEndpoint(Mockito.eq(projectId));
+            Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).doGetResponseFromCloud(Mockito.eq(endpoint),
+                    Mockito.eq(cloudUser));
+            Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).doGetAllTypesResponseFrom(Mockito.eq(json));
+            Assert.assertEquals(expected, e.getMessage());
+        }
+    }
+    
+    // test case: When calling the doGetAllTypesResponseFrom method with a JSON
+    // malformed, it must verify that a UnexpectedException was throw.
+    @Test
+    public void testDoGetAllTypesResponseFromJsonMalformed() {
+        // set up
+        String json = JSON_MALFORMED;
+        String expected = Messages.Error.ERROR_WHILE_PROCESSING_VOLUME_REQUIREMENTS;
+
+        try {
+            // exercise
+            this.plugin.doGetAllTypesResponseFrom(json);
+            Assert.fail();
+        } catch (UnexpectedException e) {
+            // verify
+            Assert.assertEquals(expected, e.getMessage());
+        }
+    }
+    
+    // test case: When calling the doGetResponseFromCloud method, it must verify
+    // that the call was successful.
     @Test
-    public void getInstanceState() {
-        // TODO
+    public void testDoGetResponseFromCloud() throws Exception {
+        // set up
+        OpenStackV3User cloudUser = createOpenStackUser();
+        String endpoint = generateEndpoint(cloudUser.getProjectId(), OpenStackVolumePlugin.VOLUMES,
+                TestUtils.FAKE_INSTANCE_ID);
+
+        String json = FAKE_VOLUME_JSON_RESPONSE;
+        Mockito.doReturn(json).when(this.client).doGetRequest(Mockito.eq(endpoint), Mockito.eq(cloudUser));
+
+        // exercise
+        this.plugin.doGetResponseFromCloud(endpoint, cloudUser);
+
+        // verify
+        Mockito.verify(this.client, Mockito.times(TestUtils.RUN_ONCE)).doGetRequest(Mockito.eq(endpoint),
+                Mockito.eq(cloudUser));
+    }
+    
+    private VolumeInstance createVolumeInstance() {
+        String id = TestUtils.FAKE_INSTANCE_ID;
+        String cloudState = AVAILABLE_STATE;
+        String name = TestUtils.FAKE_ORDER_NAME;
+        int size = TestUtils.DISK_VALUE;
+        return new VolumeInstance(id, cloudState, name, size);
+    }
+    
+    private String generateEndpoint(String projectId, String resource, String instanceId) {
+        String endpoint = PREFIX_ENDPOINT 
+                + OpenStackVolumePlugin.V2_API_ENDPOINT 
+                + projectId
+                + resource;
+
+        if (instanceId != null) {
+            return endpoint + OpenStackVolumePlugin.ENDPOINT_SEPARATOR + instanceId;
+        }
+        return endpoint;
     }
+    
+    private OpenStackV3User createOpenStackUser() {
```

Comment:
shouldn't it be in openstackCloudUtil?

Reply:
I don't understand, this is an auxiliary method for testing. Perhaps, may you want an OpenStackTestUtils helper class that contains common testing methods for openstack cloud plugins?

Reply:
I've got confused, Sorry, I mean testutils

Reply:
The createSystemUser method of the TestUtils class is not of the same type as OpenStackV3User.